service/src/main/java/org/cbioportal/service/util/CoExpressionAsyncMethods.java

```diff
@@ -0,0 +1,68 @@
+package org.cbioportal.service.util;
+
+import org.springframework.stereotype.Component;
+import org.springframework.scheduling.annotation.Async;
+import org.cbioportal.model.CoExpression;
+import org.cbioportal.service.exception.GeneNotFoundException;
+import org.cbioportal.service.exception.GenesetNotFoundException;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.stat.correlation.SpearmansCorrelation;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.math.BigDecimal;
+import java.util.concurrent.CompletableFuture;
+
+@Component
+public class CoExpressionAsyncMethods {
+
+    @Async
+    public CompletableFuture<CoExpression> computeCoExpression(String entityId, List<String> valuesA, List<String> valuesB,
+            Boolean isMolecularProfileBOfGenesetType, Double threshold, String molecularProfileId) throws GenesetNotFoundException, GeneNotFoundException {
+
+        List<String> valuesACopy = new ArrayList<>(valuesA);
+        List<String> valuesBCopy = new ArrayList<>(valuesB);
+
+        List<Integer> valuesToRemove = new ArrayList<>();
+        for (int i = 0; i < valuesBCopy.size(); i++) {
+            if (!NumberUtils.isNumber(valuesBCopy.get(i)) || !NumberUtils.isNumber(valuesACopy.get(i))) {
```

Comment:
`isNumber` is deprecated.
```suggestion
            if (!NumberUtils.isCreatable(valuesBCopy.get(i)) || !NumberUtils.isCreatable(valuesACopy.get(i))) {
```